client/components/AddProduct.js

```diff
@@ -0,0 +1,66 @@
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+import { postProduct } from '../store/product'
+
+class AddProduct extends Component {
+    constructor(props){
+        super(props)
+        this.state = {
+            title: '',
+            description: '',
+            price: '',
+            quantity: '',
+            imageUrl: ''
+        }
+        this.handleChange = this.handleChange.bind(this)
+        this.handleSubmit = this.handleSubmit.bind(this)
+    }
+
+    handleChange(event){
+        let name = event.target.name
+        let value = event.target.value
+
+        this.setState({
+            [name]: value
+        })
+    }
+
+    handleSubmit(event){
+        event.preventDefault()
+        this.props.addProduct(this.state)
+    }
+
+    render(){
+        console.log(this.state)
+        return (
+            <div>
+                <form onSubmit={this.handleSubmit}>
+                    <label>Title</label>
+                    <input name="title" onChange={this.handleChange}/>
+                    <label>Description</label>
+                    <input name="description" onChange={this.handleChange}/>
+                    <label>Price</label>
+                    <input name="price" onChange={this.handleChange}/>
+                    <label>Quantity</label>
+                    <input name="quantity" onChange={this.handleChange}/>
+                    <label>Image URL</label>
+                    <input name="imageUrl" onChange={this.handleChange}/>
+                    <button type="submit">Submit</button>
+                </form>
+            </div>
+        )
+    }
+}
+
+const mapState = ()=>({})
```

Comment:
Do we need a mapState if we dont use it?

Reply:
We still need to pass two arguments to the connect function